viral/sequence_utils.py

```
@@ -0,0 +1,859 @@
+    the principal shuffle used in the main figures involved the random re-ordering (resampling without replacement)
+    of the bins observed within a given event."
+    "'timeBinPermutation': permutes (resamples without replacement)"
+
+    Essentially, this is a Python implementation of https://github.com/losonczylab/Grosmark_NatNeuro_2021/blob/main/shufflePopulationEvents.m
+    """
+    if mode == "linear":
+        shuffled_weighted_rs = list()
+        xy = construct_xy_by_bin(
+            posterior_probability_matrix, mode="linear", total_length=total_length
+        )
+        for i in range(n_shuffles):
```

Ruff (0.14.10)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)